### PR TITLE
Some fixes to pass internal building pipeline

### PR DESCRIPTION
**Pull request by yiylin**
- [x] Does this PR have a descriptive title that could go in our release notes?
- [ ] Does this PR add any new dependencies? No
- [ ] Does this PR modify any existing APIs? No
   - [ ] Is the change to the API backwards compatible?
- [ ] Should this result in any changes to our documentation, either updating existing docs or adding new ones? No
 
#### What does this implement/fix? Briefly explain your changes.
Fix some compile errors in internal building pipeline

### CODE REVIEW

File: include/utils.h [hunk 1]
```diff
@@ -31,6 +31,9 @@ typedef int FileHandle;
 #include "memory_mapped_files.h"
 #endif
 
+#undef max
```
Comment (harsha-simhadri) [inline include/utils.h:34]: Why is this needed?

File: src/pq_flash_index.cpp [hunk 2]
```diff
@@ -1178,7 +1178,6 @@ void PQFlashIndex<T, LabelT>::cached_beam_search(const T *query1, const _u64 k_s
     unsigned cmps = 0;
     unsigned hops = 0;
     unsigned num_ios = 0;
-    unsigned k = 0;
```
Comment (harsha-simhadri) [inline src/pq_flash_index.cpp:1181]: where is this extra variable used?

File: include/utils.h (hunk 1 above)
Comment (yiylin) [reply include/utils.h:34]: it could conflict with some macro

Comment (yiylin) [reply include/utils.h:34]: https://github.com/microsoft/DiskANN/blob/main/CMakeLists.txt . here you also configure "NOMINMAX". but I could not use this way in our project, or else it will cause other compile errors

File: src/pq_flash_index.cpp (hunk 2 above)
Comment (yiylin) [reply src/pq_flash_index.cpp:1181]: no code uses this variable. our internal building pipeline catches this

File: src/distance.cpp [hunk 3]
```diff
@@ -79,8 +79,6 @@ float SlowDistanceCosineUInt8::compare(const uint8_t *a, const uint8_t *b, uint3
 
 float DistanceL2Int8::compare(const int8_t *a, const int8_t *b, uint32_t size) const
 {
-    int32_t result = 0;
```
[the remark this answers is not in the document]
Comment (yiylin) [reply src/distance.cpp:82]: In the some case(Window), this variable is defined, but no place uses it. so it causes the compiler error in our internal building pipeline

File: include/utils.h (hunk 1 above)
Comment (harsha-simhadri) [reply include/utils.h:34]: @theantony Does this change affect you? Asking since you included the -DNOMINMAX in windows cmake options.

Comment (theantony) [reply include/utils.h:35]: Thanks @harsha-simhadri ! @yiylin What is the scenario where you're encountering the issue? Are you building DiskANN as part of your pipeline and the build breaks? Or are you using pre-built DiskANN as nuget package or similar and are getting build breaks when using DiskANN headers?

Comment (yiylin) [reply include/utils.h:35]: I use this repo as submodule, then include this header to our project. Do you have the pre-built DiskANN nuget package?

Comment (theantony) [reply include/utils.h:35]: We build DiskANN as part of another nuget, it's not an "official" nuget.

I think these #undef calls are not the right place in DiskANN, they may cause issues with projects using DiskANN.

If you're building DiskANN as submodule, won't you get `NOMINMAX` defined so you won't have this issue? Or if you're using your custom build setting, can you add `NOMINMAX` just for DiskANN?

Then if you really need to #undef min or max, you can do it before including DiskANN headers and not impact other users of DiskANN.

Comment (yiylin) [reply include/utils.h:35]: fine. if you have concerns, then I could remove this. I will try other ways to fix the issue in our side.

Comment (theantony) [reply include/utils.h]: Thanks! Looks good.

**Review by theantony — APPROVED** [review]
